### PR TITLE
start with templates json files and Matlab code to create them

### PR DESCRIPTION
**Pull request by dorahermes**
This is a beginning of the starters kit template json files. There is also Matlab code that I used to create the templates.

### CODE REVIEW

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-ShortExample_run-01_bold.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "TaskName": "",
+    "RepetitionTime": ""
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-ShortExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
+    "EchoTime": "",
+    "InversionTime": "",
+    "SliceTiming": "",
+    "SliceEncodingDirection": "",
+    "NumberOfVolumesDiscardedByScanner": "",
+    "NumberOfVolumesDiscardedByUser": "",
+    "DelayTime": "",
+    "FlipAngle": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
+    "EchoTime": "",
+    "InversionTime": "",
+    "SliceTiming": "",
+    "SliceEncodingDirection": "",
+    "NumberOfVolumesDiscardedByScanner": "",
+    "NumberOfVolumesDiscardedByUser": "",
+    "DelayTime": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "ParallelReductionFactorInPlane": "",
+    "ParallelAcquisitionTechnique": "",
+    "PartialFourier": "",
+    "PartialFourierDirection": "",
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
+    "EchoTime": "",
+    "InversionTime": "",
+    "SliceTiming": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a list of numerical values not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "NumberShots": "",
+    "ParallelReductionFactorInPlane": "",
+    "ParallelAcquisitionTechnique": "",
+    "PartialFourier": "",
+    "PartialFourierDirection": "",
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
+    "EchoTime": "",
+    "InversionTime": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    "PulseSequenceDetails": "",
+    "NumberShots": "",
+    "ParallelReductionFactorInPlane": "",
+    "ParallelAcquisitionTechnique": "",
+    "PartialFourier": "",
+    "PartialFourierDirection": "",
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
+    "EchoTime": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    "PulseSequenceType": "",
+    "PulseSequenceDetails": "",
+    "NumberShots": "",
+    "ParallelReductionFactorInPlane": "",
+    "ParallelAcquisitionTechnique": "",
+    "PartialFourier": "",
+    "PartialFourierDirection": "",
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
+    "TotalReadoutTime": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    "CoilCombinationMethod": "",
+    "PulseSequenceType": "",
+    "PulseSequenceDetails": "",
+    "NumberShots": "",
+    "ParallelReductionFactorInPlane": "",
+    "ParallelAcquisitionTechnique": "",
+    "PartialFourier": "",
+    "PartialFourierDirection": "",
+    "PhaseEncodingDirection": "",
+    "EffectiveEchoSpacing": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    "DeviceSerialNumber": "",
+    "SoftwareVersions": "",
+    "ReceiveCoilName": "",
+    "GradientSetType": "",
+    "MRTransmitCoilSequence": "",
+    "MatrixCoilMode": "",
+    "CoilCombinationMethod": "",
+    "PulseSequenceType": "",
+    "PulseSequenceDetails": "",
+    "NumberShots": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmri/sub-01_ses-01_task-FullExample_run-01_bold.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case2_run-01_phase2.json [hunk 10]
```diff
@@ -0,0 +1,4 @@
+{
+    "EchoTime": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case2_run-01_phase2.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case1_run-01_phasediff.json [hunk 11]
```diff
@@ -0,0 +1,5 @@
+{
+    "EchoTime1": "",
+    "EchoTime2": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case1_run-01_phasediff.json]: This should be a numerical value not a string.

File: templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case1_run-01_phasediff.json [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{
+    "EchoTime1": "",
```
Comment (chrisfilo) [inline templates/sub-01/ses-01/fmap/sub-01_ses-01_task-Case1_run-01_phasediff.json]: This should be a numerical value not a string.

File: templates/dataset_description.json [hunk 13]
```diff
@@ -0,0 +1,11 @@
+{
+    "Name": "",
+    "BIDSVersion": "1.0.2",
+    "License": "",
+    "Authors": "",
```
Comment (chrisfilo) [inline templates/dataset_description.json]: This should be a list of strings not a single string.

File: templates/dataset_description.json [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+    "Acknowledgements": "",
+    "HowToAcknowledge": "",
+    "Funding": "",
```
Comment (chrisfilo) [inline templates/dataset_description.json]: This should be a list of strings not a single string.

File: templates/dataset_description.json [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+    "ReferencesAndLinks": "",
```
Comment (chrisfilo) [inline templates/dataset_description.json]: This should be a list of strings not a single string.

**Comment by dorahermes** [comment]
@chrisfilo thank you! I updated by fork wiht the correct number fields, should I close the pull request and do a new one? 

We do plan to link to this from the wiki.

**Comment by chrisfilo** [comment]
The pull requests was automatically updated (github is magic!).

I noticed that you denote "empty" numerical values as `[ ]`. This to me suggest a list/array. Not sure what would be the most intutive solution here. Maybe `0`? @choldgraf what do you think?

**Comment by dorahermes** [comment]
it's easy to make it 0 if that's more intuitive

**Comment by choldgraf** [comment]
Why not define them as "nan"? I don't know if 0 is a good option because then it's hard to disambiguate an actual entry of 0, vs. a missing value that's represented w/ 0.

**Comment by chrisfilo** [comment]
What about NaN without quotes? It will immediately make any parser throw an
error.

Best,
Chris

**Comment by dorahermes** [comment]
This website specifies null, but I can not easily write that with Matlab code...
https://www.ibm.com/support/knowledgecenter/SSV2LR/com.ibm.wbpm.wid.integ.doc/topics/rjsonnullunsempprops.html

NaN (without quotes) does work, and does throw an error in the online json editor. I could also write it out: "number", that will be easily picked up by the validator if left unchanged.

**Comment by chrisfilo** [comment]
NaN sounds like a good compromise.

**Comment by dorahermes** [comment]
As discussed, all empty fields will be an empty string "". This is a valide json document, but it will fail in the validator.

**Comment by dorahermes** [comment]
Anything else to fix before merging?

**Comment by chrisfilo** [comment]
LGTM

Best,
Chris

**Comment by Remi-Gau** [comment]
@all-contributors please add @dorahermes  for code content ideas

**Comment by Remi-Gau** [comment]
@all-contributors please add @dorahermes for code content ideas

**Comment by Remi-Gau** [comment]
@all-contributors please add @chrisgorgo  for review

**Comment by Remi-Gau** [comment]
@all-contributors please add @choldgraf  for review

**Pull request merged**